diff --git a/app/core/meson.build b/app/core/meson.build
index 89a205d32e..9eac808115 100644
--- a/app/core/meson.build
+++ b/app/core/meson.build
@@ -254,7 +254,6 @@ libappcore_sources = [
   appcoremarshal,
   cursors_sources,
   pickers_sources,
-  icons_core_sources,
 ]
 
 libappcore = static_library('appcore',
diff --git a/configure.ac b/configure.ac
index 58ae88b898..ab288edb5f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -86,7 +86,6 @@ m4_define([perl_required_version], [5.10.0])
 m4_define([poppler_required_version], [0.69.0])
 m4_define([poppler_data_required_version], [0.4.9])
 m4_define([python3_required_version], [3.6.0])
-m4_define([rsvg_required_version], [2.40.6])
 m4_define([vala_required_version], [0.40.0])
 m4_define([webkitgtk_required_version], [2.20.3])
 m4_define([webp_required_version], [0.6.0])
@@ -186,7 +185,6 @@ PERL_REQUIRED_VERSION=perl_required_version
 POPPLER_REQUIRED_VERSION=poppler_required_version
 POPPLER_DATA_REQUIRED_VERSION=poppler_data_required_version
 PYTHON3_REQUIRED_VERSION=python3_required_version
-RSVG_REQUIRED_VERSION=rsvg_required_version
 VALA_REQUIRED_VERSION=vala_required_version
 WEBKITGTK_REQUIRED_VERSION=webkitgtk_required_version
 WEBP_REQUIRED_VERSION=webp_required_version
@@ -221,7 +219,6 @@ AC_SUBST(PERL_REQUIRED_VERSION)
 AC_SUBST(POPPLER_REQUIRED_VERSION)
 AC_SUBST(POPPLER_DATA_REQUIRED_VERSION)
 AC_SUBST(PYTHON3_REQUIRED_VERSION)
-AC_SUBST(RSVG_REQUIRED_VERSION)
 AC_SUBST(VALA_REQUIRED_VERSION)
 AC_SUBST(WEBKITGTK_REQUIRED_VERSION)
 AC_SUBST(WEBP_REQUIRED_VERSION)
@@ -1936,22 +1933,6 @@ fi
 AM_CONDITIONAL(HAVE_WEBKIT, test "x$have_webkit" = xyes)
 
 
-###################
-# Check for librsvg
-###################
-
-PKG_CHECK_MODULES(SVG, librsvg-2.0 >= rsvg_required_version,,
-                  [add_deps_error([rsvg >= rsvg_required_version])])
-
-MIME_TYPES="$MIME_TYPES;image/svg+xml"
-
-AC_CHECK_PROGS(GTK_ENCODE_SYMBOLIC_SVG, gtk-encode-symbolic-svg, no)
-
-if test "x$GTK_ENCODE_SYMBOLIC_SVG" = xno; then
-  add_deps_error([gtk-encode-symbolic-svg],
-                 [Could not find gtk-encode-symbolic-svg in your PATH.])
-fi
-
 ####################################
 # Allow to disable the print plug-in
 ####################################
@@ -2451,6 +2432,7 @@ else
 fi
 
 AM_CONDITIONAL(BUILD_VALA, test "x$with_vala" != xno)
+AM_CONDITIONAL(ENABLE_VAPIGEN, test "x$with_vala" != xno)
 
 ###########################################################
 # Some plug-ins don't build on Win32, others are Win32-only
diff --git a/icons/Makefile.am b/icons/Makefile.am
index 0d6c34d1bb..ef36d65848 100644
--- a/icons/Makefile.am
+++ b/icons/Makefile.am
@@ -1,7 +1,6 @@
 SUBDIRS = \
 	Color		\
-	Legacy		\
-	Symbolic
+	Legacy
 
 iconsdir = $(gimpdatadir)/icons/hicolor
 
diff --git a/libgimpwidgets/meson.build b/libgimpwidgets/meson.build
index 63f086dc33..9a2261de0c 100644
--- a/libgimpwidgets/meson.build
+++ b/libgimpwidgets/meson.build
@@ -88,7 +88,6 @@ libgimpwidgets_sources = [
 
   gimpwidgetsenums,
   gimpwidgetsmarshal,
-  icons_imgs_sources,
   cursors_sources,
   pickers_sources,
 ]
diff --git a/m4macros/vapigen.m4 b/m4macros/vapigen.m4
index f2df12f23f..6e5b4d65d3 100644
--- a/m4macros/vapigen.m4
+++ b/m4macros/vapigen.m4
@@ -92,5 +92,5 @@ AC_DEFUN([VAPIGEN_CHECK],
   AC_SUBST([VAPIGEN_VAPIDIR])
   AC_SUBST([VAPIGEN_MAKEFILE])
 
-  AM_CONDITIONAL(ENABLE_VAPIGEN, test "x$enable_vala" = "xyes")
+  AM_CONDITIONAL(ENABLE_VAPIGEN, test "x$with_vala" = "xyes")
 ])
diff --git a/meson.build b/meson.build
index e20ca7d30f..b71e53b8a6 100644
--- a/meson.build
+++ b/meson.build
@@ -402,8 +402,6 @@ conf.set_quoted('MYPAINT_BRUSHES_DIR', mypaint_brushes_dir)
 pangocairo_minver = '1.42.0'
 pangocairo        = dependency('pangocairo',         version: '>='+pangocairo_minver)
 pangoft2          = dependency('pangoft2',           version: '>=1.29.4')
-rsvg_minver       = '2.40.6'
-rsvg              = dependency('librsvg-2.0',        version: '>='+rsvg_minver)
 
 conf.set('PANGO_DISABLE_DEPRECATED',pangocairo.version().version_compare('<1.43'))
 
@@ -1552,7 +1550,6 @@ install_conf.set('PANGOCAIRO_REQUIRED_VERSION',   pangocairo_minver)
 install_conf.set('POPPLER_DATA_REQUIRED_VERSION', poppler_data_minver)
 install_conf.set('POPPLER_REQUIRED_VERSION',      poppler_minver)
 install_conf.set('PYTHON3_REQUIRED_VERSION',      python3_minver)
-install_conf.set('RSVG_REQUIRED_VERSION',         rsvg_minver)
 install_conf.set('WEBKITGTK_REQUIRED_VERSION',    webkit_minver)
 install_conf.set('WEBP_REQUIRED_VERSION',         webp_minver)
 install_conf.set('WMF_REQUIRED_VERSION',          wmf_minver)
@@ -1681,7 +1678,6 @@ subdir('cursors')
 subdir('data')
 subdir('desktop')
 subdir('etc')
-subdir('icons')
 subdir('m4macros')
 subdir('menus')
 subdir('themes')
@@ -1784,7 +1780,6 @@ foreach dir : [
     { 'dir': 'libgimpmodule',  'deps': libgimpmodule},
     { 'dir': 'libgimpthumb',   'deps': libgimpthumb},
     { 'dir': 'libgimpwidgets', 'deps': libgimpwidgets},
-    { 'dir': 'icons' },
     { 'dir': 'plug-ins', },
   ]
   run_target('install-' + dir.get('dir'),
@@ -1804,7 +1799,6 @@ final_message = [
 '''Optional Features:''',
 '''  Check updates at startup:  @0@'''.format(get_option('check-update')),
 '''  Language selection:        @0@'''.format(isocodes.found()),
-'''  Vector icons:              @0@'''.format(have_vector_icons),
 '''  Dr. Mingw (Win32):         @0@'''.format(drmingw.found()),
 '''  Relocatable Bundle:        @0@'''.format(relocatable_bundle),
 '''  Default ICC directory:     @0@'''.format(icc_directory),
diff --git a/plug-ins/common/Makefile.am b/plug-ins/common/Makefile.am
index aa7dc474a3..637adacbfa 100644
--- a/plug-ins/common/Makefile.am
+++ b/plug-ins/common/Makefile.am
@@ -91,7 +91,6 @@ file_ps_libexecdir = $(gimpplugindir)/plug-ins/file-ps
 file_psp_libexecdir = $(gimpplugindir)/plug-ins/file-psp
 file_raw_data_libexecdir = $(gimpplugindir)/plug-ins/file-raw-data
 file_sunras_libexecdir = $(gimpplugindir)/plug-ins/file-sunras
-file_svg_libexecdir = $(gimpplugindir)/plug-ins/file-svg
 file_tga_libexecdir = $(gimpplugindir)/plug-ins/file-tga
 file_wmf_libexecdir = $(gimpplugindir)/plug-ins/file-wmf
 file_xbm_libexecdir = $(gimpplugindir)/plug-ins/file-xbm
@@ -168,7 +167,6 @@ file_ps_libexec_PROGRAMS = $(FILE_PS)
 file_psp_libexec_PROGRAMS = file-psp
 file_raw_data_libexec_PROGRAMS = file-raw-data
 file_sunras_libexec_PROGRAMS = file-sunras
-file_svg_libexec_PROGRAMS = file-svg
 file_tga_libexec_PROGRAMS = file-tga
 file_wmf_libexec_PROGRAMS = $(FILE_WMF)
 file_xbm_libexec_PROGRAMS = file-xbm
@@ -1049,27 +1047,6 @@ file_sunras_LDADD = \
 	$(INTLLIBS)		\
 	$(file_sunras_RC)
 
-file_svg_CFLAGS = $(SVG_CFLAGS)
-
-file_svg_SOURCES = \
-	file-svg.c
-
-file_svg_LDADD = \
-	$(libgimpui)		\
-	$(libgimpwidgets)	\
-	$(libgimpmodule)	\
-	$(libgimp)		\
-	$(libgimpmath)		\
-	$(libgimpconfig)	\
-	$(libgimpcolor)		\
-	$(libgimpbase)		\
-	$(GTK_LIBS)		\
-	$(GEGL_LIBS)		\
-	$(SVG_LIBS)		\
-	$(RT_LIBS)		\
-	$(INTLLIBS)		\
-	$(file_svg_RC)
-
 file_tga_SOURCES = \
 	file-tga.c
 
diff --git a/plug-ins/common/gimprc.common b/plug-ins/common/gimprc.common
index c17fe2991b..43d1592c38 100644
--- a/plug-ins/common/gimprc.common
+++ b/plug-ins/common/gimprc.common
@@ -43,7 +43,6 @@ file_ps_RC = file-ps.rc.o
 file_psp_RC = file-psp.rc.o
 file_raw_data_RC = file-raw-data.rc.o
 file_sunras_RC = file-sunras.rc.o
-file_svg_RC = file-svg.rc.o
 file_tga_RC = file-tga.rc.o
 file_wmf_RC = file-wmf.rc.o
 file_xbm_RC = file-xbm.rc.o
diff --git a/plug-ins/common/meson.build b/plug-ins/common/meson.build
index 8b41c30478..fd8ffa5fe3 100644
--- a/plug-ins/common/meson.build
+++ b/plug-ins/common/meson.build
@@ -55,9 +55,6 @@ common_plugins_list = [
   },
   { 'name': 'file-raw-data', },
   { 'name': 'file-sunras', },
-  { 'name': 'file-svg',
-    'deps': [ gtk3, gegl, rsvg, ],
-  },
   { 'name': 'file-tga', },
   { 'name': 'file-xbm', },
   { 'name': 'file-xwd', },
diff --git a/plug-ins/common/plugin-defs.pl b/plug-ins/common/plugin-defs.pl
index e6f94316b9..6786fcb4a5 100644
--- a/plug-ins/common/plugin-defs.pl
+++ b/plug-ins/common/plugin-defs.pl
@@ -44,7 +44,6 @@
     'file-psp' => { ui => 1, gegl => 1, libs => 'Z_LIBS' },
     'file-raw-data' => { ui => 1, gegl => 1 },
     'file-sunras' => { ui => 1, gegl => 1 },
-    'file-svg' => { ui => 1, gegl => 1, libs => 'SVG_LIBS', cflags => 'SVG_CFLAGS' },
     'file-tga' => { ui => 1, gegl => 1 },
     'file-wmf' => { ui => 1, gegl => 1, optional => 1, libs => 'WMF_LIBS', cflags => 'WMF_CFLAGS' },
     'file-xbm' => { ui => 1, gegl => 1 },
