diff --git a/fc-blanks/fc-blanks.py b/fc-blanks/fc-blanks.py
index 410178a..1ae31ac 100755
--- a/fc-blanks/fc-blanks.py
+++ b/fc-blanks/fc-blanks.py
@@ -2,43 +2,12 @@
 
 from __future__ import absolute_import
 from __future__ import print_function
-try:
-    from urllib2 import urlopen
-    from urllib2 import URLError
-except ImportError:
-    from urllib.request import urlopen
-    from urllib.error import URLError
 
 import sys
 import os
-from lxml import html
 from six.moves import range
 
-datafile = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'list-unicodeset.html')
-try:
-    fp = urlopen('http://unicode.org/cldr/utility/list-unicodeset.jsp?a=[%3AGC%3DZs%3A][%3ADI%3A]&abb=on&ucd=on&esc=on&g')
-    data = fp.read()
-    fp.close()
-    # check before writing if data isn't corrupted.
-    dom = html.fromstring(data)
-    x = dom.xpath('/html/body/form/p/text()')
-    p = x[1]
-    fp = open(datafile, 'wb');
-    fp.write(data);
-    fp.close();
-except (URLError, IndexError):
-    # fall back reading the static data in repo
-    try:
-        fp = open(datafile)
-        data = fp.read()
-        fp.close()
-    except IOError:
-        sys.stderr.write("Error: No static data to generate the blank data. please make sure the network connection is reachable to Unicode.org\n")
-        sys.exit(1)
-
-dom = html.fromstring(data)
-x = dom.xpath('/html/body/form/p/text()')
-p = x[1]
+p = '[\\ \\u00A0\\u00AD\\u034F\\u061C\\u115F\\u1160\\u1680\\u17B4\\u17B5\\u180B-\\u180E\\u2000-\\u200F\\u202A-\\u202F\\u205F-\\u206F\\u3000\\u3164\\uFE00-\\uFE0F\\uFEFF\\uFFA0\\uFFF0-\\uFFF8\\U0001BCA0-\\U0001BCA3\\U0001D173-\\U0001D17A\\U000E0000-\\U000E0FFF]'
 if p[0] == '[' and p[-1] == ']':
     p = p.replace('[', '').replace(']', '')
 else:
@@ -127,7 +96,7 @@ ncode = 0
 for i in l:
     ncode += (i[1] - i[0] + 1)
 
-a = int(x[0].split(' ')[0].replace(',', ''))
+a = 4190
 if a != ncode:
     sys.stderr.write("Unexpected the amount of code points: %d (expected %d)\n" % (ncode, a))
     sys.exit(1)
