diff --git a/NEWS b/NEWS
index b4288e9b..6ad1c377 100644
--- a/NEWS
+++ b/NEWS
@@ -1,6 +1,22 @@
-GNU tar NEWS - User visible changes. 2021-01-07
+GNU tar NEWS - User visible changes. 2021-02-13
 Please send GNU tar bug reports to <bug-tar@gnu.org>
 
+version 1.34 - Sergey Poznyakoff, 2021-02-13
+
+* Fix extraction over pipe (savannah bug #60002)
+
+* Fix memory leak in read_header (savannah bug #59897)
+
+* Fix extraction when . and .. are unreadable
+
+See https://lists.gnu.org/archive/html/bug-tar/2021-01/msg00012.html
+
+* Gracefully handle duplicate symlinks when extracting
+
+See https://lists.gnu.org/archive/html/bug-tar/2021-01/msg00026.html
+
+* Re-initialize supplementary groups when switching to user privileges
+
 version 1.33 - Sergey Poznyakoff, 2021-01-07
 
 * POSIX extended format headers do not include PID by default
diff --git a/configure.ac b/configure.ac
index 8beba5d1..1b9d34e1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,7 +17,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-AC_INIT([GNU tar], [1.33], [bug-tar@gnu.org])
+AC_INIT([GNU tar], [1.34], [bug-tar@gnu.org])
 AC_CONFIG_SRCDIR([src/tar.c])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_HEADERS([config.h])
