diff --git a/src/cli_options.cc b/src/cli_options.cc
index 4440e832..31a8bc6f 100644
--- a/src/cli_options.cc
+++ b/src/cli_options.cc
@@ -48,7 +48,7 @@ FluxboxCli::Options::Options() : xsync(false) {
         session_display.assign(env);
     }
 
-    string fname = std::string("~/.") + realProgramName("fluxbox");
+    string fname = std::string("~/.config/") + realProgramName("fluxbox");
     rc_path = FbTk::StringUtil::expandFilename(fname);
 
     if (!rc_path.empty()) {
diff --git a/src/tests/menutest.cc b/src/tests/menutest.cc
index 0008ff02..ff2ef7c6 100644
--- a/src/tests/menutest.cc
+++ b/src/tests/menutest.cc
@@ -135,7 +135,7 @@ private:
 
 int main(int argc, char **argv) {
     string displayname("");
-    string stylefile, menufile="~/.fluxbox/menu";
+    string stylefile, menufile="~/.config/fluxbox/menu";
     for (int a=1; a<argc; ++a) {
         if (strcmp("-display", argv[a]) == 0 && a + 1 < argc) {
             displayname = argv[++a];
diff --git a/util/fbrun/main.cc b/util/fbrun/main.cc
index 16ca9239..3c68a9e8 100644
--- a/util/fbrun/main.cc
+++ b/util/fbrun/main.cc
@@ -64,7 +64,7 @@ void showUsage(const char *progname) {
         "   -fg [color name]            Foreground text color"<<endl<<
         "   -bg [color name]            Background color"<<endl<<
         "   -na                         Disable antialias"<<endl<<
-        "   -hf [history file]          History file to load (default ~/.fluxbox/fbrun_history)"<<endl<<
+        "   -hf [history file]          History file to load (default ~/.config/fluxbox/fbrun_history)"<<endl<<
         "   -help                       Show this help"<<endl<<endl<<
         "Example: fbrun -fg black -bg white -text xterm -title \"run xterm\""<<endl;
 }
@@ -82,7 +82,7 @@ int main(int argc, char **argv) {
     string foreground("black");   // text color
     string background("white");   // text background color
     string display_name; // name of the display connection
-    string history_file("~/.fluxbox/fbrun_history"); // command history file
+    string history_file("~/.config/fluxbox/fbrun_history"); // command history file
     // parse arguments
     for (int i=1; i<argc; i++) {
         string arg = argv[i];
diff --git a/util/fbsetbg b/util/fbsetbg
index e2773171..6ad131d3 100644
--- a/util/fbsetbg
+++ b/util/fbsetbg
@@ -45,7 +45,7 @@
 
 # The wallpapersetter is selected in this order
 wpsetters="${wpsetters:=Esetroot wmsetbg feh hsetroot chbg display qiv xv xsri xli xsetbg}" # broken icewmbg'
-lastwallpaper="${HOME}/.fluxbox/lastwallpaper"
+lastwallpaper="${HOME}/.config/fluxbox/lastwallpaper"
 
 
 WHOAMI=`whoami`
@@ -99,7 +99,7 @@ Options:
 
 Files:
 
-    ~/.fluxbox/lastwallpaper     In this file the wallpaper you set
+    ~/.config/fluxbox/lastwallpaper     In this file the wallpaper you set
                                  will be stored, for the -l option:
 Environment variables:
     wpsetters   Wallpapersetters to use.
@@ -123,7 +123,7 @@ Common tips to use with $command:
    perl -pi -e 's,([^f]|^)bsetbg,$command,' filename
 
 2) If you want to set the wallpaper and you want $command to
-   remember the previous wallpaper put this in your ~/.fluxbox/overlay
+   remember the previous wallpaper put this in your ~/.config/fluxbox/overlay
 
    background: none
 
@@ -194,8 +194,8 @@ if [ $# -eq 0 ]; then
 fi
 
 # create directory and last wallpaper file 
-if [ ! -d  "$HOME/.fluxbox" ]; then
-	mkdir "$HOME/.fluxbox"
+if [ ! -d  "$HOME/.config/fluxbox" ]; then
+	mkdir "$HOME/.config/fluxbox"
     touch ${lastwallpaper}
 fi
 
diff --git a/util/fluxbox-generate_menu.in b/util/fluxbox-generate_menu.in
index 085d6222..611a654e 100755
--- a/util/fluxbox-generate_menu.in
+++ b/util/fluxbox-generate_menu.in
@@ -1220,7 +1220,7 @@ case ${LC_ALL} in
 esac
 
 # Set Defaults
-USERFLUXDIR="${HOME}/.@pkgprefix@fluxbox@pkgsuffix@"
+USERFLUXDIR="${HOME}/.config/@pkgprefix@fluxbox@pkgsuffix@"
 MENUFILENAME="${MENUFILENAME:=${USERFLUXDIR}/menu}"
 MENUTITLE="${MENUTITLE:=Fluxbox}"
 HOMEPAGE="${HOMEPAGE:=fluxbox.org}"
diff --git a/util/fluxbox-update_configs.cc b/util/fluxbox-update_configs.cc
index 4065f976..cd10f908 100644
--- a/util/fluxbox-update_configs.cc
+++ b/util/fluxbox-update_configs.cc
@@ -99,7 +99,7 @@ void update_add_mouse_evens_to_keys(FbTk::ResourceManager& rm,
 void update_move_groups_entries_to_apps_file(FbTk::ResourceManager& rm,
         const FbTk::FbString& keyfilename, const FbTk::FbString& appsfilename) {
 
-    FbTk::Resource<string> rc_groupfile(rm, "~/.fluxbox/groups",
+    FbTk::Resource<string> rc_groupfile(rm, "~/.config/fluxbox/groups",
             "session.groupFile", "Session.GroupFile");
     string groupfilename = FbTk::StringUtil::expandFilename(*rc_groupfile);
     string whole_groupfile = read_file(groupfilename);
@@ -540,9 +540,9 @@ const Update UPDATES[] = {
 int run_updates(int old_version, FbTk::ResourceManager &rm) {
     int new_version = old_version;
 
-    FbTk::Resource<string> rc_keyfile(rm, "~/.fluxbox/keys",
+    FbTk::Resource<string> rc_keyfile(rm, "~/.config/fluxbox/keys",
             "session.keyFile", "Session.KeyFile");
-    FbTk::Resource<string> rc_appsfile(rm, "~/.fluxbox/apps",
+    FbTk::Resource<string> rc_appsfile(rm, "~/.config/fluxbox/apps",
             "session.appsFile", "Session.AppsFile");
 
     string appsfilename = FbTk::StringUtil::expandFilename(*rc_appsfile);
@@ -600,7 +600,7 @@ int main(int argc, char **argv) {
     }
 
     if (rc_filename.empty())
-        rc_filename = FbTk::StringUtil::expandFilename("~/.fluxbox/init");
+        rc_filename = FbTk::StringUtil::expandFilename("~/.config/fluxbox/init");
 
     FbTk::ResourceManager resource_manager(rc_filename.c_str(),false);
     if (!resource_manager.load(rc_filename.c_str())) {
diff --git a/util/startfluxbox.in b/util/startfluxbox.in
index 3c083f36..4cac3381 100755
--- a/util/startfluxbox.in
+++ b/util/startfluxbox.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 command="`basename \"$0\"`"
-fluxdir="$HOME/.@pkgprefix@fluxbox@pkgsuffix@"
+fluxdir="$HOME/.config/@pkgprefix@fluxbox@pkgsuffix@"
 startup="$fluxdir/startup"
 
 while [ $# -gt 0 ]; do
