diff --git a/player/lua.c b/player/lua.c
index 98dcfee5b2..74ee8363c6 100644
--- a/player/lua.c
+++ b/player/lua.c
@@ -373,8 +373,8 @@ static int load_lua(struct mpv_handle *client, const char *fname)
         .filename = fname,
     };
 
-    if (LUA_VERSION_NUM != 501 && LUA_VERSION_NUM != 502) {
-        MP_FATAL(ctx, "Only Lua 5.1 and 5.2 are supported.\n");
+    if (LUA_VERSION_NUM != 501 && LUA_VERSION_NUM != 502 && LUA_VERSION_NUM != 503) {
+        MP_FATAL(ctx, "Only Lua 5.1, 5.2 and 5.3 are supported.\n");
         goto error_out;
     }
 
diff --git a/player/lua/osc.lua b/player/lua/osc.lua
index de492152b1..85ed5e9101 100644
--- a/player/lua/osc.lua
+++ b/player/lua/osc.lua
@@ -134,7 +134,9 @@ end
 
 function set_virt_mouse_area(x0, y0, x1, y1, name)
     local sx, sy = get_virt_scale_factor()
-    mp.set_mouse_area(x0 / sx, y0 / sy, x1 / sx, y1 / sy, name)
+    if sx > 0 and sy > 0 then
+        mp.set_mouse_area(x0 // sx, y0 // sy, x1 // sx, y1 // sy, name)
+    end
 end
 
 function scale_value(x0, x1, y0, y1, val)
@@ -521,7 +523,7 @@ function render_elements(master_ass)
         end
 
         style_ass:append(string.format("{\\1a&H%X&\\2a&H%X&\\3a&H%X&\\4a&H%X&}",
-            ar[1], ar[2], ar[3], ar[4]))
+            math.floor(ar[1]), math.floor(ar[2]), math.floor(ar[3]), math.floor(ar[4])))
 
         if element.eventresponder and (state.active_element == n) then
 
@@ -2118,8 +2120,8 @@ function render()
     end
 
     -- submit
-    mp.set_osd_ass(osc_param.playresy * osc_param.display_aspect,
-                   osc_param.playresy, ass.text)
+    mp.set_osd_ass(math.floor(osc_param.playresy * osc_param.display_aspect),
+                   math.floor(osc_param.playresy), ass.text)
 
 
 
diff --git a/waftools/checks/custom.py b/waftools/checks/custom.py
index 821bf9c46f..49bbb526cf 100644
--- a/waftools/checks/custom.py
+++ b/waftools/checks/custom.py
@@ -66,6 +66,8 @@ def check_lua(ctx, dependency_identifier):
         ( '52deb',  'lua5.2 >= 5.2.0'), # debian
         ( '52fbsd', 'lua-5.2 >= 5.2.0'), # FreeBSD
         ( 'luajit', 'luajit >= 2.0.0' ),
+        ( '53',     'lua >= 5.3.0 lua < 5.4.0' ),
+        ( '53deb',  'lua5.3 >= 5.3.0'), # debian
     ]
 
     if ctx.options.LUA_VER:
