diff --git a/defs/gmake.mk b/defs/gmake.mk
index f093fd4e26..16697e6733 100644
--- a/defs/gmake.mk
+++ b/defs/gmake.mk
@@ -260,22 +260,11 @@ $(srcdir)/.bundle/cache:
 
 gems/%.gem:
 	$(ECHO) Downloading bundled gem $*...
-	$(Q) $(BASERUBY) -C "$(srcdir)" \
-	    -I./tool -rdownloader \
-	    -e 'gem = "$(@F)"' \
-	    -e 'old = Dir.glob("gems/"+gem.sub(/-[^-]*$$/, "-*.gem"))' \
-	    -e 'Downloader::RubyGems.download(gem, "gems", nil) and' \
-	    -e '(old.delete("gems/#{gem}"); !old.empty?) and' \
-	    -e 'File.unlink(*old) and' \
-	    -e 'FileUtils.rm_rf(old.map{'"|n|"'n.chomp(".gem")})'
 
 extract-gems: | $(patsubst %,.bundle/gems/%,$(bundled-gems))
 
 .bundle/gems/%: gems/%.gem | .bundle/gems
 	$(ECHO) Extracting bundle gem $*...
-	$(Q) $(BASERUBY) -C "$(srcdir)" \
-	    -Itool -rgem-unpack \
-	    -e 'Gem.unpack("gems/$(@F).gem", ".bundle/gems")'
 
 $(srcdir)/.bundle/gems:
 	$(MAKEDIRS) $@
