diff --git a/gen_tzinfo.py b/gen_tzinfo.py
index d06fa0e..a88211a 100644
--- a/gen_tzinfo.py
+++ b/gen_tzinfo.py
@@ -52,6 +52,8 @@ def links():
         # Filenames containing a '.' are not data files.
         if '.' in os.path.basename(filename):
             continue
+        if filename in ('tz/tzselect', 'tz/yearistype', 'tz/zdump', 'tz/zic', 'tz/NEWS'):
+            continue
         for line in open(filename):
             if line.strip().startswith('#') or not line.strip():
                 continue
@@ -123,20 +125,20 @@ def add_allzones(filename):
             cz.append(zone)
     cz.sort()
 
-    print >> outf, 'all_timezones = \\'
+    print('all_timezones = \\', file=outf)
     pprint(sorted(allzones()), outf)
-    print >> outf, '''all_timezones = LazyList(
+    print ('''all_timezones = LazyList(
         tz for tz in all_timezones if resource_exists(tz))
-        '''
-    print >> outf, 'all_timezones_set = LazySet(all_timezones)'
-    print >> outf, '_all_timezones_lower_to_standard = dict((tz.lower(), tz) for tz in all_timezones)'
+        ''', file=outf)
+    print('all_timezones_set = LazySet(all_timezones)', file=outf)
+    print('_all_timezones_lower_to_standard = dict((tz.lower(), tz) for tz in all_timezones)', file=outf)
 
-    print >> outf, 'common_timezones = \\'
+    print('common_timezones = \\', file=outf)
     pprint(cz, outf)
-    print >> outf, '''common_timezones = LazyList(
+    print('''common_timezones = LazyList(
             tz for tz in common_timezones if tz in all_timezones)
-        '''
-    print >> outf, 'common_timezones_set = LazySet(common_timezones)'
+        ''', file=outf)
+    print('common_timezones_set = LazySet(common_timezones)', file=outf)
 
     outf.close()
 
