diff --git a/offlineimap/bundled_imaplib2.py b/offlineimap/bundled_imaplib2.py
index 1fd47d2..7c01ba5 100755
--- a/offlineimap/bundled_imaplib2.py
+++ b/offlineimap/bundled_imaplib2.py
@@ -302,8 +302,8 @@ class IMAP4(object):
 
 
     # These must be encoded according to utf8 setting in _mode_xxx():
-    _literal = br'.*{(?P<size>\d+)}$'
-    _untagged_status = br'\* (?P<data>\d+) (?P<type>[A-Z-]+)( (?P<data2>.*))?'
+    _literal = r'.*{(?P<size>\d+)}$'
+    _untagged_status = r'\* (?P<data>\d+) (?P<type>[A-Z-]+)( (?P<data2>.*))?'
 
     continuation_cre = re.compile(r'\+( (?P<data>.*))?')
     mapCRLF_cre = re.compile(r'\r\n|\r|\n')
diff --git a/offlineimap/folder/Maildir.py b/offlineimap/folder/Maildir.py
index f061bcd..ec9853d 100644
--- a/offlineimap/folder/Maildir.py
+++ b/offlineimap/folder/Maildir.py
@@ -74,7 +74,7 @@ class MaildirFolder(BaseFolder):
         # Everything up to the first comma or colon (or ! if Windows):
         self.re_prefixmatch = re.compile('([^'+ self.infosep + ',]*)')
         # folder's md, so we can match with recorded file md5 for validity.
-        self._foldermd5 = md5(self.getvisiblename()).hexdigest()
+        self._foldermd5 = md5(self.getvisiblename().encode()).hexdigest()
         # Cache the full folder path, as we use getfullname() very often.
         self._fullname = os.path.join(self.getroot(), self.getname())
         # Modification time from 'Date' header.
diff --git a/offlineimap/imapserver.py b/offlineimap/imapserver.py
index 600ce65..b4a89ef 100644
--- a/offlineimap/imapserver.py
+++ b/offlineimap/imapserver.py
@@ -215,8 +215,8 @@ class IMAPServer(object):
         # At this point all authz, authc and passwd are expected bytes encoded
         # in UTF-8.
         NULL = b'\x00'
-        retval = NULL.join((authz, authc, passwd))
-        logsafe_retval = NULL.join((authz, authc, b'(passwd hidden for log)'))
+        retval = NULL.join((authz, authc.encode(), passwd.encode()))
+        logsafe_retval = NULL.join((authz, authc.encode(), b'(passwd hidden for log)'))
         self.ui.debug('imap', '__plainhandler: returning %s'% logsafe_retval)
         return retval
 
diff --git a/offlineimap/utils/distro.py b/offlineimap/utils/distro.py
index 87bb2b1..f4140f7 100644
--- a/offlineimap/utils/distro.py
+++ b/offlineimap/utils/distro.py
@@ -70,7 +70,7 @@ def get_os_sslcertfile_searchpath():
     l = None
     if OS in __DEF_OS_LOCATIONS:
         l = __DEF_OS_LOCATIONS[OS]
-        if not hasattr(l, '__iter__'):
+        if isinstance(l, str):
             l = (l, )
     return l
 
