diff -Naur sqlite-autoconf-3080900/shell.c sqlite-autoconf-3080900-new/shell.c
--- sqlite-autoconf-3080900/shell.c	2015-04-08 17:38:50.000000000 +0300
+++ sqlite-autoconf-3080900-new/shell.c	2015-04-10 03:25:17.534619247 +0300
@@ -4678,9 +4678,9 @@
       }
       zHome = find_home_dir();
       if( zHome ){
-        nHistory = strlen30(zHome) + 20;
+        nHistory = strlen30(zHome) + 26;
         if( (zHistory = malloc(nHistory))!=0 ){
-          sqlite3_snprintf(nHistory, zHistory,"%s/.sqlite_history", zHome);
+          sqlite3_snprintf(nHistory, zHistory,"%s/.cache/sqlite_history", zHome);
         }
       }
       if( zHistory ) shell_read_history(zHistory);
