diff --git a/kitty/data-types.h b/kitty/data-types.h
index 51dbb16a..9f85d753 100644
--- a/kitty/data-types.h
+++ b/kitty/data-types.h
@@ -309,9 +309,6 @@ void fake_scroll(int, bool);
 void set_special_key_combo(int glfw_key, int mods, bool is_native);
 void on_key_input(int key, int scancode, int action, int mods, const char*, int);
 void request_window_attention(id_type, bool);
-#ifndef __APPLE__
-void play_canberra_sound(const char *which_sound, const char *event_id);
-#endif
 SPRITE_MAP_HANDLE alloc_sprite_map(unsigned int, unsigned int);
 SPRITE_MAP_HANDLE free_sprite_map(SPRITE_MAP_HANDLE);
 
diff --git a/kitty/desktop.c b/kitty/desktop.c
index 3348aab4..5681ae4b 100644
--- a/kitty/desktop.c
+++ b/kitty/desktop.c
@@ -7,7 +7,6 @@
 
 #include "data-types.h"
 #include <dlfcn.h>
-#include <canberra.h>
 
 #define FUNC(name, restype, ...) typedef restype (*name##_func)(__VA_ARGS__); static name##_func name = NULL
 #define LOAD_FUNC(handle, name) {\
@@ -90,25 +89,9 @@ static PyMethodDef module_methods[] = {
     {NULL, NULL, 0, NULL}        /* Sentinel */
 };
 
-static ca_context *canberra_ctx = NULL;
-
-void
-play_canberra_sound(const char *which_sound, const char *event_id) {
-    if (canberra_ctx == NULL) ca_context_create(&canberra_ctx);
-    ca_context_play(
-        canberra_ctx, 0,
-        CA_PROP_EVENT_ID, which_sound,
-        CA_PROP_EVENT_DESCRIPTION, event_id,
-        NULL
-    );
-}
-
 static void
 finalize(void) {
     if (libsn_handle) dlclose(libsn_handle);
-    libsn_handle = NULL;
-    if (canberra_ctx) ca_context_destroy(canberra_ctx);
-    canberra_ctx = NULL;
 }
 
 bool
diff --git a/kitty/glfw.c b/kitty/glfw.c
index b9de8de3..54345395 100644
--- a/kitty/glfw.c
+++ b/kitty/glfw.c
@@ -842,13 +842,9 @@ ring_audio_bell(OSWindow *w UNUSED) {
     double now = monotonic();
     if (now - last_bell_at <= 0.1) return;
     last_bell_at = now;
-#ifdef __APPLE__
     if (w->handle) {
         glfwWindowBell(w->handle);
     }
-#else
-    play_canberra_sound("bell", "kitty bell");
-#endif
 }
 
 static PyObject*
diff --git a/setup.py b/setup.py
index e7c6ba93..41990073 100755
--- a/setup.py
+++ b/setup.py
@@ -270,9 +270,6 @@ def kitty_env():
     gl_libs = ['-framework', 'OpenGL'] if is_macos else pkg_config('gl', '--libs')
     libpng = pkg_config('libpng', '--libs')
     ans.ldpaths += pylib + font_libs + gl_libs + libpng
-    if not is_macos:
-        cflags.extend(pkg_config('libcanberra', '--cflags-only-I'))
-        ans.ldpaths += pkg_config('libcanberra', '--libs')
     if is_macos:
         ans.ldpaths.extend('-framework Cocoa'.split())
     else:
