From d7a1cd14b7916e301e9719c44a8167fcb9d3565f Mon Sep 17 00:00:00 2001
From: Aleksey Rybalkin <aleksey@rybalkin.org>
Date: Fri, 23 Jan 2015 20:38:12 +0300
Subject: [PATCH] Issue 309: hide cursor on EnterNotify

---
 src/x11_window.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/x11_window.c b/src/x11_window.c
index 0380b58..5caa900 100644
--- a/src/x11_window.c
+++ b/src/x11_window.c
@@ -1054,6 +1054,9 @@ static void processEvent(XEvent *event)
 
         case EnterNotify:
         {
+            if (window->cursorMode == GLFW_CURSOR_HIDDEN)
+                hideCursor(window);
+
             _glfwInputCursorEnter(window, GL_TRUE);
             break;
         }
-- 
2.3.0

