From 5a7ea8a05e347754aa15508a2466017279ed3d9a Mon Sep 17 00:00:00 2001
From: Aleksey Rybalkin <aleksey@rybalkin.org>
Date: Fri, 23 Jan 2015 20:38:12 +0300
Subject: [PATCH] Issue 309: hide cursor on EnterNotify and LeaveNotify

---
 src/x11_window.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/x11_window.c b/src/x11_window.c
index 0380b58..6835345 100644
--- a/src/x11_window.c
+++ b/src/x11_window.c
@@ -1054,12 +1054,18 @@ static void processEvent(XEvent *event)
 
         case EnterNotify:
         {
+            if (window->cursorMode == GLFW_CURSOR_HIDDEN)
+                hideCursor(window);
+
             _glfwInputCursorEnter(window, GL_TRUE);
             break;
         }
 
         case LeaveNotify:
         {
+            if (window->cursorMode == GLFW_CURSOR_HIDDEN)
+                hideCursor(window);
+
             _glfwInputCursorEnter(window, GL_FALSE);
             break;
         }
-- 
2.2.2

