diff --git a/background_scripts/main.coffee b/background_scripts/main.coffee
index 1aef5f33..55106115 100644
--- a/background_scripts/main.coffee
+++ b/background_scripts/main.coffee
@@ -453,6 +453,8 @@ sendRequestHandlers =
     chrome.tabs.create url: chrome.runtime.getURL("pages/options.html"), index: request.tab.index + 1
   frameFocused: handleFrameFocused
   nextFrame: BackgroundCommands.nextFrame
+  copyToClipboard: Clipboard.copy.bind Clipboard
+  pasteFromClipboard: Clipboard.paste.bind Clipboard
   selectSpecificTab: selectSpecificTab
   createMark: Marks.create.bind(Marks)
   gotoMark: Marks.goto.bind(Marks)
diff --git a/content_scripts/link_hints.coffee b/content_scripts/link_hints.coffee
index 231f65ef..9b55eeb2 100644
--- a/content_scripts/link_hints.coffee
+++ b/content_scripts/link_hints.coffee
@@ -33,7 +33,7 @@ COPY_LINK_URL =
     if link.href?
       url = link.href
       url = url[7..] if url[...7] == "mailto:"
-      HUD.copyToClipboard url
+      chrome.runtime.sendMessage handler: "copyToClipboard", data: url
       url = url[0..25] + "...." if 28 < url.length
       HUD.showForDuration "Yanked #{url}", 2000
     else
diff --git a/content_scripts/mode_visual.coffee b/content_scripts/mode_visual.coffee
index 4c6578cd..f99e42f9 100644
--- a/content_scripts/mode_visual.coffee
+++ b/content_scripts/mode_visual.coffee
@@ -312,7 +312,7 @@ class VisualMode extends KeyHandlerMode
   yank: (args = {}) ->
     @yankedText = @selection.toString()
     @exit()
-    HUD.copyToClipboard @yankedText
+    chrome.runtime.sendMessage handler: "copyToClipboard", data: @yankedText
 
     message = @yankedText.replace /\s+/g, " "
     message = message[...12] + "..." if 15 < @yankedText.length
diff --git a/manifest.json b/manifest.json
index 328c5994..7ecf83ba 100644
--- a/manifest.json
+++ b/manifest.json
@@ -13,6 +13,7 @@
       "lib/settings.js",
       "background_scripts/bg_utils.js",
       "background_scripts/commands.js",
+      "lib/clipboard.js",
       "background_scripts/exclusions.js",
       "background_scripts/completion_engines.js",
       "background_scripts/completion_search.js",
diff --git a/pages/help_dialog.coffee b/pages/help_dialog.coffee
index 08180a72..f36155e4 100644
--- a/pages/help_dialog.coffee
+++ b/pages/help_dialog.coffee
@@ -83,7 +83,7 @@ HelpDialog =
               commandNameElement.textContent = command.command
               commandNameElement.title = "Click to copy \"#{command.command}\" to clipboard."
               commandNameElement.addEventListener "click", ->
-                HUD.copyToClipboard commandNameElement.textContent
+                chrome.runtime.sendMessage handler: "copyToClipboard", data: commandNameElement.textContent
                 HUD.showForDuration("Yanked #{commandNameElement.textContent}.", 2000)
 
       @showAdvancedCommands(@getShowAdvancedCommands())
