diff -r b4c82b704180 mercurial/worker.py
--- a/mercurial/worker.py	Wed Dec 18 20:17:21 2019 +0100
+++ b/mercurial/worker.py	Fri Feb 07 22:53:36 2020 +0300
@@ -204,7 +204,7 @@
                         os.close(w)
                     os.close(rfd)
                     for result in func(*(staticargs + (pargs,))):
-                        os.write(wfd, util.pickle.dumps(result))
+                        os.write(wfd, util.pickle.dumps(result, protocol=3))
                     return 0
 
                 ret = scmutil.callcatch(ui, workerfunc)
