diff --git a/bootstrap.py b/bootstrap.py
index 8fa9e4b5..4e4856ad 100644
--- a/bootstrap.py
+++ b/bootstrap.py
@@ -48,6 +48,8 @@ def add_minimal_info():
     os.mkdir('setuptools.egg-info')
     with io.open('setuptools.egg-info/entry_points.txt', 'w') as ep:
         ep.write(minimal_egg_info)
+    with io.open('setuptools.egg-info/PKG-INFO', 'w') as pi:
+        pi.write('Version: {VERSION}')
 
 
 def run_egg_info():
