diff --git a/ui/gl/gl_surface_glx.cc b/ui/gl/gl_surface_glx.cc
index 52261c1dbf23..8afea79bf1d7 100644
--- a/ui/gl/gl_surface_glx.cc
+++ b/ui/gl/gl_surface_glx.cc
@@ -459,10 +459,11 @@ bool GLSurfaceGLX::InitializeOneOff() {
     return false;
   }
 
-  auto* visual_picker = gl::GLVisualPickerGLX::GetInstance();
-  XVisualInfo visual_info = visual_picker->rgba_visual();
-  if (!visual_info.visual)
-    visual_info = visual_picker->system_visual();
+  const auto& visual_info =
+      ui::IsCompositingManagerPresent()
+          ? gl::GLVisualPickerGLX::GetInstance()->rgba_visual()
+          : gl::GLVisualPickerGLX::GetInstance()->system_visual();
+
   g_visual = visual_info.visual;
   g_depth = visual_info.depth;
   g_colormap =
